test/windows/WSLATests.cpp

```diff
@@ -1098,4 +1136,112 @@ class WSLATests
     }
+
+    TEST_METHOD(CreateContainer)
+    {
+        WSL2_TEST_ONLY();
+
+#ifdef _ARM64_
+
+        LogSkipped("Skipping CreateContainer test case for ARM64");
+        return;
+
+#else
+
+        auto storageVhd = std::filesystem::current_path() / "storage.vhdx";
+
+        // Create a 1G temporary VHD.
+        if (!std::filesystem::exists(storageVhd))
+        {
+            wsl::core::filesystem::CreateVhd(storageVhd.native().c_str(), 1024 * 1024 * 1024, nullptr, true, false);
+        }
+
+        auto cleanup = wil::scope_exit_log(WI_DIAGNOSTICS_INFO, [&]() { LOG_IF_WIN32_BOOL_FALSE(DeleteFileW(storageVhd.c_str())); });
+
+        VIRTUAL_MACHINE_SETTINGS settings{};
+        settings.CpuCount = 4;
+        settings.DisplayName = L"WSLA";
+        settings.MemoryMb = 2048;
+        settings.BootTimeoutMs = 30 * 1000;
+
+        auto installedVhdPath =
+            std::filesystem::path(wsl::windows::common::wslutil::GetMsiPackagePath().value()) / L"wslarootfs.vhd";
+
+#ifdef WSL_DEV_INSTALL_PATH
+
+        settings.RootVhd = TEXT(WSLA_TEST_DISTRO_PATH);
+
+#else
+
+        settings.RootVhd = installedVhdPath.c_str();
+
+#endif
+
+        settings.RootVhdType = "squashfs";
+        settings.NetworkingMode = WSLANetworkingModeNAT;
+        settings.ContainerRootVhd = storageVhd.c_str();
+        settings.FormatContainerRootVhd = true;
+
+        auto session = CreateSession(settings);
+
+        // TODO: Remove once the proper rootfs VHD is available.
+        ExpectCommandResult(session.get(), {"/etc/lsw-init.sh"}, 0);
+
+        // Test a simple container start.
+        {
+            WSLAContainerLauncher launcher("debian:latest", "test-simple", "echo", {"OK"});
+            auto container = launcher.Launch(*session);
+            auto process = container.GetInitProcess();
+
+            ValidateProcessOutput(process, {{1, "OK\n"}});
+        }
+
+        // Validate that env is correctly wired.
+        {
+            WSLAContainerLauncher launcher("debian:latest", "test-env", "/bin/bash", {"-c", "echo $testenv"}, {{"testenv=testvalue"}});
+            auto container = launcher.Launch(*session);
+            auto process = container.GetInitProcess();
+
+            ValidateProcessOutput(process, {{1, "testvalue\n"}});
+        }
+
+        // Validate that starting containers works with the default entrypoint.
+        {
+            WSLAContainerLauncher launcher(
+                "debian:latest", "test-default-entrypoint", "/bin/cat", {}, {}, ProcessFlags::Stdin | ProcessFlags::Stdout | ProcessFlags::Stderr);
+
+            // For now, validate that trying to use stdin without a tty returns the appropriate error.
+            auto result = wil::ResultFromException([&]() { auto container = launcher.Launch(*session); });
+
+            VERIFY_ARE_EQUAL(result, HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED));
+
+            // This is hanging. nerdctl run seems to hang with -i is passed outside of a TTY context.
```

Comment:
Just /bin/cat with -i works via shell for me, the only hang I see is with redirection to /dev/null. Curious why the above is hanging here.